backend/lambdas/process_post/lambda.py

```diff
@@ -0,0 +1,217 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: MIT-0
+
+import json
+import logging
+import os
+import boto3
+import demoji
+import re
+
+from langchain.prompts import PromptTemplate
+from langchain.llms.bedrock import Bedrock
+from langchain_community.chat_models import BedrockChat
+from langchain_core.output_parsers import StrOutputParser
+from langchain_core.prompts import ChatPromptTemplate
+
+from text_preprocessing import TweetPreprocessor
+from llm_prompts_by_lang import prompts_map, examples_map, json_format_str
+
+model_id = os.environ['MODEL_ID']
+language_code = os.environ['LANGUAGE_CODE']
+meta_topics = os.environ['LABELS'].split(',')
+meta_sentiments = os.environ['SENTIMENT_LABELS'].split(',')
+
+meta_sentiments = [sentiment.strip() for sentiment in meta_sentiments if sentiment.strip() != '']
+meta_topics = [topic.strip() for topic in meta_topics if topic.strip() != '']
+
+bedrock_runtime = boto3.client(
+    service_name="bedrock-runtime",
+    region_name="us-east-1"
+)
+
+model_id_mapping = {
+    'claude': 'anthropic.claude-3-haiku-20240307-v1:0',
+}
+
+logging.getLogger().setLevel(os.environ.get('LOG_LEVEL', 'WARNING').upper())
```

Comment:
Have you considered using Powertools for Lambda? They offer a cool [structured logging](https://docs.powertools.aws.dev/lambda/python/latest/core/logger/) convenience.

Reply:
Fixed in commit [e611804](https://github.com/aws-samples/ai-powered-text-insights/pull/3/commits/e6118046a160abba4d131ed45237e7d3c6391261)